x-pack/plugins/endpoint/server/routes/resolver/utils/fetch.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IScopedClusterClient } from 'kibana/server';
+import { extractEntityID, extractParentEntityID } from './normalize';
+import { getPaginationParams } from './pagination';
+import { Tree } from './tree';
+import { LifecycleQuery } from '../queries/lifecycle';
+import { ChildrenQuery } from '../queries/children';
+import { EventsQuery } from '../queries/events';
+import { StatsQuery } from '../queries/stats';
+
+export class Fetcher {
+  constructor(
+    private readonly client: IScopedClusterClient,
+    private readonly id: string,
+    private readonly endpointID?: string
+  ) {}
+
+  public async ancestors(limit: number): Promise<Tree> {
```

Comment:
These functions would make more sense as functions of the `Tree` class imo. The original PR with some additional comments is here: https://github.com/elastic/kibana/pull/58172

Reply:
@marshallmain are you ok with us addressing your feedback in a follow up PR later?

Reply:
I think we should address in a follow up PR